crates/context_aware_config/src/api/custom_types/mod.rs

```diff
@@ -0,0 +1,159 @@
+    path: Path<String>,
+    db_conn: DbConnection,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let _ = JSONSchema::compile(&request).map_err(|err| {
+        log::error!(
+            "Invalid jsonschema sent in the request, schema: {:?} error: {}",
+            request,
+            err
+        );
+        bad_argument!(
+            "Invalid jsonschema sent in the request, validation error is: {}",
+            err.to_string()
+        )
+    })?;
+    let type_name = path.into_inner();
+    let regex = Regex::new(TYPE_NAME_REGEX).map_err(|err| {
+        unexpected_error!("could not parse regex due to: {}", err.to_string())
+    })?;
+    if !regex.is_match(type_name.as_str()) {
+        return Err(bad_argument!(
+            "The type name {} is invalid, it should obey the regex {}",
+            type_name,
+            TYPE_NAME_REGEX
+        ));
+    }
+    let timestamp = Utc::now().naive_utc();
+    let updated_type = diesel::update(jsonschema_types::table)
+        .filter(jsonschema_types::type_name.eq(type_name.clone()))
+        .set((
+            jsonschema_types::type_schema.eq(request.clone()),
+            jsonschema_types::created_by.eq(user.email),
+            jsonschema_types::last_modified.eq(timestamp),
+        ))
+        .get_result::<JsonSchemaTypes>(&mut conn)
+        .map_err(|err| {
+            log::error!("failed to insert custom type with error: {}", err);
+            db_error!(err)
+        })?;
+    Ok(HttpResponse::Ok().json(updated_type))
+}
+
+#[delete("/{type_name}")]
+async fn delete_type(
+    path: Path<String>,
+    db_conn: DbConnection,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let type_name = path.into_inner();
+    let deleted_type =
+        diesel::delete(dsl::jsonschema_types.filter(dsl::type_name.eq(type_name)))
+            .get_result::<JsonSchemaTypes>(&mut conn)?;
+    Ok(HttpResponse::Ok().json(deleted_type))
+}
+
+#[get("/{page}/{count}")]
+async fn list_types(
+    db_conn: DbConnection,
+    filters: Path<(i64, i64)>,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let (page, limit) = filters.into_inner();
+    if page < 1 {
+        return Err(bad_argument!("Param 'page' has to be at least 1"));
+    } else if limit < 1 {
+        return Err(bad_argument!("Param 'count' has to be at least 1"));
+    }
+    let n_types: i64 = jsonschema_types::dsl::jsonschema_types
+        .count()
+        .get_result(&mut conn)?;
+    let offset = (page - 1) * limit;
+    let custom_types: Vec<JsonSchemaTypes> = jsonschema_types::dsl::jsonschema_types
+        .order(jsonschema_types::dsl::created_at.desc())
+        .limit(limit)
+        .offset(offset)
+        .load(&mut conn)?;
+    let total_pages = (n_types as f64 / limit as f64).ceil() as u64;
+    Ok(HttpResponse::Ok().json(json!({
+        "total_pages": total_pages,
```

Comment:
@pratikmishra356 any paginated list API should have these two fields along so that the UI can paginate easily rather than guessing how many pages are left. extra DB call is fine, count is fast